### PR TITLE
FHIR updates

### PR DESCRIPTION
**Pull request by joeflack4**
Updates
```
    FHIR
    - OboGraphToFHIRConverter: Added 'retention' param (Include all edges, or just some specific ones?) to relevant methods.
    - Misc: Minor codestyle / docstring updates.

    General
    - .gitignore: Added .DS_Store. Organized into sections: (i) directories, (ii) dot files, (iii) related dirs/files/patterns
```

Also includes:
- https://github.com/INCATools/ontology-access-kit/pull/373

Has some updates for:
- https://github.com/INCATools/ontology-access-kit/issues/369

### CODE REVIEW

File: src/oaklib/converters/obo_graph_to_fhir_converter.py [hunk 1]
```diff
@@ -48,23 +53,29 @@
 class OboGraphToFHIRConverter(DataModelConverter):
     """Converts from OboGraph to FHIR."""
 
-    def dump(self, source: GraphDocument, target: str = None) -> None:
+    # TODO: `retention` is temporary. I think we likely have to give this a different name/functionality, as well as put
```
Comment (joeflack4) [inline src/oaklib/converters/obo_graph_to_fhir_converter.py:56]: `retention` param (see comment)

File: src/oaklib/converters/obo_graph_to_fhir_converter.py [hunk 2]
```diff
@@ -88,29 +100,32 @@ def code(self, uri: CURIE) -> str:
...
+        _id = self.code(source.id)
+        logging.debug(f"Converting node {_id} from {source}")
+        concept = Concept(code=_id, display=source.lbl)
         target.concept.append(concept)
         if source.meta:
             self._convert_meta(source, concept)
         for e in index.get(source.id, []):
             obj = self.code(e.obj)
-            if e.pred in DIRECT_PREDICATE_MAP:
+            if retention == 'maximal' or e.pred in DIRECT_PREDICATE_MAP:
```
Comment (joeflack4) [inline src/oaklib/converters/obo_graph_to_fhir_converter.py:126]: Usage of `retention` param

File: src/oaklib/converters/obo_graph_to_fhir_converter.py [hunk 3]
```diff
@@ -48,23 +53,29 @@
 class OboGraphToFHIRConverter(DataModelConverter):
     """Converts from OboGraph to FHIR."""
 
-    def dump(self, source: GraphDocument, target: str = None) -> None:
+    # TODO: `retention` is temporary. I think we likely have to give this a different name/functionality, as well as put
+    #  it in the CLI and possibly elsewhere. I'm also doing this a pure way, passing it to dump and then further down,
+    #  rather than an instance property. - Joe 2022/12/01
+    def dump(self, source: GraphDocument, target: str = None, retention: str = ['maximal', 'minimal'][0]) -> None:
```
Comment (joeflack4) [inline src/oaklib/converters/obo_graph_to_fhir_converter.py:59]: I also got `src/oaklib/converters/obo_graph_to_fhir_converter.py:59:14: BLK100 Black would make changes.`
I can't see what it recommends, but it's probably `[0]`. I can just default the string and add the valid params to the docstring, but I like to do this in a more formal way outside of comments and more in something like a type system, but I don't actually know how you do /if you can do that in Python (tell it which param values are valid).

File: src/oaklib/converters/obo_graph_to_fhir_converter.py (hunk 1 above)
Comment (joeflack4) [reply src/oaklib/converters/obo_graph_to_fhir_converter.py]: Changed to `include_all_predicates` (bool)

File: src/oaklib/converters/obo_graph_to_fhir_converter.py (hunk 3 above)
Comment (joeflack4) [reply src/oaklib/converters/obo_graph_to_fhir_converter.py]: I spoke w/ Harshad and I ran `black` manually

File: src/oaklib/cli.py [hunk 4]
```diff
@@ -1849,8 +1849,14 @@ def descendants(terms, predicates, display: str, output_type: str, output: TextI
...
 @click.option("-o", "--output")
+@click.option(
+    "--include-all-predicates/--no-include-all-predicates",
+    default=False,
+    show_default=True,
+    help="For formats that export only IS_A by default, this will include all possible predicates",
+)
 @output_type_option
-def dump(terms, output, output_type: str):
+def dump(terms, output, output_type: str, **kwargs):
```
Comment (cmungall) [inline src/oaklib/cli.py:1859]: @joeflack4 - note here how we pass through generically all other click options. For now this is only include_all_predicates, but if more options are added, they would be passed through

File: src/oaklib/interfaces/dumper_interface.py [hunk 5]
```diff
@@ -27,7 +27,7 @@ class DumperInterface(BasicOntologyInterface, ABC):
     An OntologyInterface that is capable of exporting entire contents.
     """
 
-    def dump(self, path: str = None, syntax: str = None):
+    def dump(self, path: str = None, syntax: str = None, **kwargs):
```
Comment (cmungall) [inline src/oaklib/interfaces/dumper_interface.py:30]: @joeflack4 - note how the *interface* accepts an open-ended list of params. Different syntaxes, and different implementations may have different such params

**Review by cmungall — APPROVED** [review]

**Comment by joeflack4** [comment]
@cmungall I _super_ appreciate these timely merges!

File: src/oaklib/cli.py (hunk 4 above)
Comment (joeflack4) [reply src/oaklib/cli.py:1859]: I saw this earlier; looks good! Thanks much!

**Pull request merged**